### PR TITLE
[pointer_interceptor] Adopt code-excerpt in README

### PR DESCRIPTION
**Pull request by foxtrotravi**
Improves README and updates it to use code excerpts.

Part of [#102679](https://github.com/flutter/flutter/issues/102679)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/pointer_interceptor/pointer_interceptor/README.md [hunk 1]
```diff
@@ -44,26 +45,46 @@ All the cases above have in common that they attempt to render Flutter widgets *
...
-    ```
+<?code-excerpt "main.dart (PointerInterceptorWrapper)"?>
+```dart
+PointerInterceptor(
+  child: ElevatedButton(
+    key: const Key('clickable-button'),
+    child: const Text('Works As Expected'),
+    onPressed: () {
+      _clickedOn('clickable-button');
+    },
```
Comment (ditman) [inline packages/pointer_interceptor/pointer_interceptor/README.md:56]: All this extra code makes the README example more complicated than what it needs to be.

You can remove unnecessary code from an excerpt, and render "..." in between the gaps, by doing something like this:

```
// #docregion PointerInterceptorWrapper
PointerInterceptor(
  child: ElevatedButton(
// #enddocregion PointerInterceptorWrapper
     key: const Key('clickable-button'),
     child: const Text('Works As Expected'),
     onPressed: () {
      _clickedOn('clickable-button');
    },
// #docregion PointerInterceptorWrapper
  ),
),
// #enddocregion PointerInterceptorWrapper
```

That way the example will still be based in the code, but it'll be more similar to the previous version!

---

The string used to "join" different parts of the example is called `plaster`, and can be configured with something like:

```
<?code-excerpt "main.dart (PointerInterceptorWrapper)" plaster="!!!"?>
```

The [default `plaster` value]() seems to be `···` (which I think is fine!):
 https://github.com/flutter/packages/blob/8163731c489338377154eb3fde2cfdd9293373c0/script/tool/lib/src/update_excerpts_command.dart#L154

---

(This applies to all generated excerpts in this file!)

Comment (foxtrotravi) [reply packages/pointer_interceptor/pointer_interceptor/README.md:56]: That's awesome. I will update this right away.

Comment (foxtrotravi) [reply packages/pointer_interceptor/pointer_interceptor/README.md]: Done.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@foxtrotravi Are you still planning on updating this PR to address the review comments above?

**Comment by stuartmorgan** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**